src/main/java/com/philips/swcoe/cerberus/cerebellum/codemetrics/java/CodeMetricsVerticalWriterService.java

```diff
@@ -14,59 +14,56 @@
-                                    .andThen(codeMetricsResultForClass -> Try.run(() -> writeClassMetrics(csvPrinter, javaCodeMetricsClassResult, codeMetricsResultForClass ))
-                                    .onFailure(exception -> log.trace(exception.getStackTrace())));
-            });
-            List<CodeMetricsMethodResult> codeMetricsMethodResults = javaCodeMetricsClassResult.getMethodMetrics();
+        codeMetricsClassResults.stream().forEach(codeMetricsClassResult -> {
+            processMetrics(codeMetricsClassResult, classConfig);
+            List<CodeMetricsMethodResult> codeMetricsMethodResults = codeMetricsClassResult.getMethodMetrics();
             codeMetricsMethodResults.stream().forEach(codeMetricsMethodResult -> {
-                getStreamMetricsMethods(codeMetricsMethodResult.getClass().getMethods()).forEach(methodInMethodResult -> {
-                    Try.of(() -> (CodeMetricsResult) methodInMethodResult.invoke(codeMetricsMethodResult))
-                                    .andThen(codeMetricsResultForMethod -> Try.run(() -> writeMethodMetrics(csvPrinter, codeMetricsMethodResult, codeMetricsResultForMethod, javaCodeMetricsClassResult))
-                                    .onFailure(exception -> log.trace(exception.getStackTrace())));
-                });
+                processMetrics(codeMetricsMethodResult, methodConfig);
             });
         });
         return this.getReport();
     }
 
-    private void writeClassMetrics(CSVPrinter csvPrinter,
-                                   CodeMetricsClassResult codeMetricsClassResult,
-                                   CodeMetricsResult codeMetricsResult) throws IOException {
-        List<String> classMetricsToDisplay = getMetricsToDisplayFromConfig(classConfig);
-        if(doesItMatterToDisplay(codeMetricsResult, classMetricsToDisplay)) {
-            csvPrinter.printRecord(
-                    codeMetricsClassResult.getFile(),
-                    codeMetricsClassResult.getClassName(),
-                    codeMetricsClassResult.getType(),
-                    codeMetricsResult.getMetricName(),
-                    codeMetricsResult.getNewValue(),
-                    codeMetricsResult.getOldValue()
-            );
+    private void processMetrics(CodeMetricsResult codeMetricsResult, List<String> config) {
+        getStreamMetricsMethods(codeMetricsResult.getClass().getMethods()).forEach(methodInResult -> {
+            Try.of(() -> (CodeMetricsDiffResult) methodInResult.invoke(codeMetricsResult))
+                                .andThen(codeMetricsResultForConstruct -> Try.run(() -> writeMetrics(codeMetricsResultForConstruct, config))
+                                .onFailure(exception -> log.trace(exception.getStackTrace())));
+        });
+    }
+
+    private void writeMetrics(CodeMetricsDiffResult codeMetricsDiffResult, List<String> config) throws IOException {
+        if(null == csvPrinter) {
```

Comment:
Nope Writer is different, and Generator is different, To write CSV or PSV we use CSV writer , during the process of instantiation of this object by birth it will possess the required writer. 
using the writer that it possess, we write the metrics. Post writing the metrics. We generate a report. Based on the writer we generate the report . 

In our context think like this, writer is the pen, and generator is the one which ensures things written or embosed on paper.  Depending on the pen that you use generator chooses the paper and embose the content on to it.